### ISSUE TITLE
Ignore `relatedTo*` arguments in GraphQL when values are `null`

### ISSUE THREAD

**Issue by joshuabaker** [craftcms/cms#7810]
Would it be feasible for GraphQL queries to automatically exclude `null` values from `relatedTo*` queries, or provide a comparable approach to reduce client side logic.

As an example, I have 3 articles. Querying them directly will return all 3 articles, as expected.

```gql
{
  entries(section: "articles") {
    id
  }
}
```

```json
{
  "data": {
    "entries": [
      {
        "id": "1"
      },
      {
        "id": "2"
      },
      {
        "id": "3"
      }
    ]
  }
}
```

When filtering this same section by `relatedToCategories` it filters the articles to only those related to a category, despite the arguments being `null` or empty.

```gql
{
  entries(section: "articles", relatedToCategories: [{group: null}]) {
    id
  }
}

#  or

{
  entries(section: "articles", relatedToCategories: [{}]) {
    id
  }
}

#  or

{
  entries(section: "articles", relatedToCategories: {}) {
    id
  }
}
```

```json
{
  "data": {
    "entries": [
      {
        "id": "1"
      },
      {
        "id": "2"
      }
    ]
  }
}
```

I would imagine these can just be passed through an array filter, effectively removing the constraint.

**Comment by joshuabaker** [comment]
Interestingly, I note that the `categories` query works this way when passing `null`.

```gql
{
  a: categories(group: "cities", relatedToEntries: [{section: null}]) {
    title
  }
  b: categories(group: "cities") {
    title
  }
}
```

Both `a` and `b` queries yield the same result here.

**Comment by joshuabaker** [comment]
This looks to have been resolved via a7542e0f7103f7ddc8784606d6e917db1d475afa (#7824).
